src/linter/ui5Types/SourceFileLinter.ts

```diff
@@ -529,10 +529,9 @@ export default class SourceFileLinter {
 			if (isDataType) {
 				this.#reporter.addMessage({
 					node: moduleSpecifierNode,
-					severity: LintMessageSeverity.Error,
-					ruleId: RULES["ui5-linter-no-pseudo-modules"],
-					message: formatMessage(MESSAGES.SHORT__NO_DIRECT_DATATYPE_ACCESS, moduleSpecifierNode.text),
-					messageDetails: formatMessage(MESSAGES.DETAILS__NO_DIRECT_DATATYPE_ACCESS, moduleNamespaceName),
+					message: MESSAGE.NO_DIRECT_DATATYPE_ACCESS,
+					args: [moduleSpecifierNode.text],
```

Comment:
What bothers me here is that we will just proxy those to the `formatMessage`.
The needed information is already available here (inside `MESSAGE.NO_DIRECT_DATATYPE_ACCESS`), so can't we somehowe directly apply arguments on top of it via the `formatMessage`?

Reply:
Do you mean to again call `formatMessage` in here?
I'm not in favour of that for two reasons.
1. it would again be a place where things could be out of sync (as at least the message + the enum needs to be provided then)
2. (more important) the memory usage could not be improved because the final message would always be calculated during linting.